### DISCUSSION TITLE
MaxRedirects for network error retry

### DISCUSSION THREAD

**Discussion by bagusardians** [redis/go-redis#2055] [Q&A]
Hi all,
Currently im using the latest go-redis, specifically the cluster. From what i know, go-redis has **MaxRetries**, to set how many times it can retry to call redis. My plan is to set the Retries to 0, which means it will just fail fast when there is any error, and goes to DB. But, when there is a network error, it still retries using the **MaxRedirects**. Based on the comments, it seems intentional. May i know, what is the reason the network error still retried using MaxRedirects? And is there any configuration alternative if i dont want to do retry on network error, but i still want to use MaxRedirects for MOVED error related.
Thank you

**Comment by vmihailenco** [comment]
From what I see, `MaxRetries` is disabled for ClusterClient and we only use `MaxRedirects`. It makes to retry network errors in case there are multiple replicas or the failing node was replaced but the cluster configuration had not been updated yet.
